test/TestCase.cpp

```diff
@@ -51,9 +51,10 @@ void TestCase::printUpdatedSettings(std::ostream& _stream, std::string const& _l
 bool TestCase::isTestFilename(boost::filesystem::path const& _filename)
 {
 	std::string extension = _filename.extension().string();
-	return (extension == ".sol" || extension == ".yul" || extension == ".stack") &&
-		   !boost::starts_with(_filename.string(), "~") &&
-			!boost::starts_with(_filename.string(), ".");
+	// NOTE: .asmjson rather than .json because JSON files that do not represent test cases exist in some test dirs.
+	return (extension == ".sol" || extension == ".yul" || extension == ".asm" || extension == ".asmjson" || extension == ".stack") &&
+		!boost::starts_with(_filename.string(), "~") &&
+		!boost::starts_with(_filename.string(), ".");
```

Comment:
```suggestion
		!_filename.string().starts_with('~') &&
		!_filename.string().starts_with('.');
```

We could use the [STL starts_with implementation](https://en.cppreference.com/w/cpp/string/basic_string/starts_with) and also use its overload for single characters